pynopegl-utils/pynopegl_utils/examples/timeranges.py

```diff
@@ -30,32 +30,21 @@ def parallel_playback(cfg: SceneCfg, fast=True, segment_time=2.0, constrained_ti
     render1 = ngl.Group(children=(render1, ngl.Text("media #1", **text_settings)))
     render2 = ngl.Group(children=(render2, ngl.Text("media #2", **text_settings)))
 
-    rf1 = ngl.TimeRangeFilter(render1)
-    rf2 = ngl.TimeRangeFilter(render2)
-
-    if constrained_timeranges:
-        rf1.set_prefetch_time(segment_time / 3.0)
-        rf2.set_prefetch_time(segment_time / 3.0)
-        rf1.set_max_idle_time(segment_time / 2.0)
-        rf2.set_max_idle_time(segment_time / 2.0)
-
-    t = 0
+    end_time = 0.0
     rr1 = []
     rr2 = []
-    while t < cfg.duration:
-        rr1.append(ngl.TimeRangeModeCont(t))
-        rr1.append(ngl.TimeRangeModeNoop(t + segment_time))
-
-        rr2.append(ngl.TimeRangeModeNoop(t))
-        rr2.append(ngl.TimeRangeModeCont(t + segment_time))
-
-        t += 2 * segment_time
-
-    rf1.add_ranges(*rr1)
-    rf2.add_ranges(*rr2)
+    i = 0
+    while end_time < cfg.duration:
+        end_time = (i + 1) * segment_time
+        rr, render = (rr2, render2) if i & 1 else (rr1, render1)
```

Comment:
nit: I find the new logic a bit harder to read.

Reply:
I added parenthesis around the destination tuple and with your suggestions the code looks sane to me:

```python
    end_time = 0.0
    rr1 = []
    rr2 = []
    i = 0
    while end_time < cfg.duration:
        (rr, render) = (rr2, render2) if i & 1 else (rr1, render1)
        start_time = i * segment_time
        end_time = (i + 1) * segment_time
        tseg = ngl.TimeRangeFilter(render, start_time, end_time)
        if constrained_timeranges:
            tseg.set_prefetch_time(segment_time / 3)
        rr.append(tseg)
        i += 1
```